### PR TITLE
Change version info

### PR DESCRIPTION
**Pull request by varnson**
Fix https://github.com/hashicorp/consul-esm/issues/87
Now we have the version like this:

> [linux_amd64]$ ./consul-esm -version
> consul-esm 0.5.0-dev (428d995)
> Compatible with Consul versions >= 1.4.1

### CODE REVIEW

**Comment by lornasong** [comment]
@varnson - thanks for your work here! I have some clarifying questions that I asked on the original issue - feel free to look there for updates.

**Review by lornasong — COMMENTED** [review]
@varnson thanks again for working on this issue. The cli output looks good for `VersionPrerelease = "dev"`. One minor thing, we  want to add a 'v' in front of the version number. For example, 'consul-esm 0.5.0 (...)' change to 'consul-esm v0.5.0 (...)'

From discussion on the [original issue](https://github.com/hashicorp/consul-esm/issues/87), this change should also work for a released binary. That is, for `VersionPrerelease = ""`, the result should be something like `consul-esm 0.5.0 (428d995)`. Currently the output is like: `v0.5.0-3-g428d995`

Let me know if you have any questions. Thanks!

File: Makefile [hunk 1]
```diff
@@ -31,9 +31,9 @@ XC_EXCLUDE ?= darwin/arm solaris/386 solaris/arm windows/arm netbsd/arm
 LD_FLAGS ?= \
 	-s \
 	-w \
-	-X ${PROJECT}/version.Name=${NAME} \
-	-X ${PROJECT}/version.GitCommit=${GIT_COMMIT} \
-	-X ${PROJECT}/version.GitDescribe=${GIT_DESCRIBE}
+	-X github.com/hashicorp/consul-esm/version.Name=${NAME} \
+	-X github.com/hashicorp/consul-esm/version.GitCommit=${GIT_COMMIT} \
+	-X github.com/hashicorp/consul-esm/version.GitDescribe=${GIT_DESCRIBE}
```
[the remark this answers is not in the document]
Comment (varnson) [reply Makefile:36]: I use go mod to  compile it, and my path is not github.com/hashicorp/consul-esm, so the $(PROJECT) is not github.com/hashicorp/consul-esm, and GitCommit will not be set.

[the remark this answers is not in the document]
Comment (lornasong) [reply Makefile:36]: I understand. Thank you for explaining! I also saw that other projects use a hard-coded string for 'ldflags'

**Comment by varnson** [comment]
Now the version looks like this:
With VersionPrerelease:
consul-esm v0.5.0-dev (428d995)
Compatible with Consul versions >= 1.4.1

Without VersionPrerelease:
consul-esm v0.5.0
Compatible with Consul versions >= 1.4.1

**Comment by lornasong** [comment]
@varnson thanks for the changes! This is almost there. The output with 'VersionPrerelease' and adding the 'v' looks great. For the output without 'VersionPrerelease', we still want to include the commit sha. Sorry for any miscommunication. Let me know if you have any questions. Thank you.

Example of the output we want:
With  'VersionPrerelease': `consul-esm v0.3.3-dev (17c7ce4)`
Without 'VersionPrerelease': `consul-esm v0.5.0 (da2279)`

**Review by lornasong — APPROVED** [review]
The version output looks great @varnson! Thanks for all your work here!

**Pull request merged**